src/Animal.js

```diff
@@ -1,12 +1,113 @@
-import { Organism } from './organism.js';
+import { Organism } from './Organism';
+import { findNearestTiles } from './findNearestTiles';
+import { classesList } from './classesList';
+import { findRandomTileInArray } from './findRandomTileInArray';
 
 export class Animal extends Organism {
-    constructor() {
-        super();
-        this.strength = 0;
+  constructor(board) {
+    super(board);
+    this.strength = 0;
+  }
+
+  mate(newTile, organism) {
+    if (newTile.currentOrganism) {
+      if (
+        organism.constructor.name === newTile.currentOrganism.constructor.name
```

Comment:
I think you can just check the constructor reference
```suggestion
        organism.constructor=== newTile.currentOrganism.constructor
```